From 317d12ee37ed0e60ea29f4456ad29856ac0c9015 Mon Sep 17 00:00:00 2001
From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Thu, 6 Oct 2022 16:50:47 +0200
Subject: [PATCH 87/92] backend-vnc: use configured keymap

Use the keymap rules/model/layout configured in the weston.ini [keyboard]
section.

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
(cherry picked from commit 2f0be4b4d0971fc876768d0e38dcd34d24f0960f)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/backend-vnc/vnc.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libweston/backend-vnc/vnc.c b/libweston/backend-vnc/vnc.c
index dbcb3950..190fe92a 100644
--- a/libweston/backend-vnc/vnc.c
+++ b/libweston/backend-vnc/vnc.c
@@ -965,9 +965,9 @@ vnc_backend_create(struct weston_compositor *compositor,
 
 	compositor->capabilities |= WESTON_CAP_ARBITRARY_MODES;
 
-	backend->xkb_rule_name.rules = strdup("evdev");
-	backend->xkb_rule_name.model = strdup("pc105");
-	backend->xkb_rule_name.layout = strdup("us");
+	backend->xkb_rule_name.rules = strdup(compositor->xkb_names.rules);
+	backend->xkb_rule_name.model = strdup(compositor->xkb_names.model);
+	backend->xkb_rule_name.layout = strdup(compositor->xkb_names.layout);
 
 	backend->xkb_keymap = xkb_keymap_new_from_names(
 					backend->compositor->xkb_context,
-- 
2.20.1

